oauthlib/oauth1/rfc5849/signature.py

```diff
@@ -134,26 +140,27 @@ def base_string_uri(uri, host=None):
     #
     # .. _`RFC3986`: https://tools.ietf.org/html/rfc3986
 
-    if not scheme or not netloc:
```

Comment:
why is this no longer the case?

Reply:
Test for missing `scheme` error is still done in the new line 143. But the `netloc` obtained from _urlparse_ **can** be empty if an overriding value is provided in the `host` parameter. It is only an error if **both** _netloc_ is empty and the _host_ is empty: see new line 200.